src/foam/core/ContextFactory.java

```diff
@@ -0,0 +1,12 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.core;
+
+public interface ContextFactory<T> {
```

Comment:
We also have XFactory: https://github.com/foam-framework/foam2/blob/f383a842c33d46772cdc81ddd13b89eb0490221a/src/foam/core/XFactory.java

Context is called X everywhere else.

Reply:
Yeah this is a little different than XFactory and I'm not sure what else to call this. Do you have any suggestions? XFObjectFactory?